src/index.ts

```diff
@@ -274,4 +354,24 @@ export class UltravoxSession {
       this.state.dispatchEvent(new UltravoxExperimentalMessageEvent(msg));
     }
   }
+
+  private getRoleSet(roles: RoleString | RoleString[] | Set<RoleString> | Role | Role[] | Set<Role>): Set<RoleString> {
```

Comment:
This seems like a lot of added complication for something that's easy to do on the consumer side.  I think it's best to only take in a Set<Role>.  A Role[] would also be okay (I think arrays are more common in JS), but that's an easy conversion since the Set constructor can take an array as input.